sfall/Modules/BugFixes.cpp

```diff
@@ -1687,6 +1754,10 @@ void BugFixes::init()
 		SafeWrite8(0x4583E5, 0x90);
 		dlogr(" Done", DL_INIT);
 	}
+
+	// Fix code: Wrong item was pass to the function, to check the perk of the object
+	HookCall(0x42954B, ai_best_weapon_hook);
```

Comment:
Just noticed an interesting glitch: while the fix itself is fairly logical, it would make Frank Horrigan equip his end boss knife (which has weapon penetrate perk) instead of the usual choice of his plasma gun (no weapon perk), so the end fight becomes much easier. Oh well, it's not the first time the devs made mistake on things like this. Giving the plasma gun some unimportant weapon perk like weapon night sight does solve it.

Reply:
There an imbalance in the calculation of ai_best_weapon, if the weapon has perk, then the priority points are multiplied by 5, IMHO, this is a lot, with respect to weapons without perk.

Reply:
![2018-08-28_03-03-02](https://user-images.githubusercontent.com/21695363/44693206-4295b780-aa6f-11e8-8308-e855dc2282bf.png)